GenderPayGap.WebUI/Models/ManageOrganisations/ManageOrganisationViewModel.cs

```diff
@@ -101,36 +106,125 @@ public string GetRequiredToReportOrNotText()
                 : "NOT REQUIRED TO REPORT";
         }
 
-        public string GetByReportingDeadlineText()
+        private DateTime GetAccountingDate()
         {
-            OrganisationScope scopeForYear = organisation.GetScopeForYear(ReportingYear);
+            return organisation.SectorType.GetAccountingStartDate(ReportingYear);
+        }
+
+        private DateTime GetReportingDeadline()
+        {
+            DateTime snapshotDate = GetAccountingDate();
+            return ReportingYearsHelper.GetDeadlineForAccountingDate(snapshotDate);
+        }
+
+        private bool DeadlineHasPassed()
+        {
+            return VirtualDateTime.Now > GetReportingDeadline();
+        }
+
+        private bool OrganisationIsRequiredToSubmit()
+        {
+            return organisation.GetScopeForYear(ReportingYear).IsInScopeVariant()
+                   && !Global.ReportingStartYearsToExcludeFromLateFlagEnforcement.Contains(GetAccountingDate().Year);
+        }
+
+        private ReportStatus GetReportStatus()
+        {
+            Return returnForYear = organisation.GetReturn(ReportingYear);
+            bool reportIsNotSubmitted = returnForYear == null;
 
-            if (scopeForYear.IsInScopeVariant())
+            if (OrganisationIsRequiredToSubmit() && reportIsNotSubmitted && !DeadlineHasPassed())
             {
-                DateTime snapshotDate = organisation.SectorType.GetAccountingStartDate(ReportingYear);
-                DateTime deadline = ReportingYearsHelper.GetDeadlineForAccountingDate(snapshotDate);
+                return ReportStatus.Due;
+            }
 
-                return "by " + deadline.ToString("d MMM yyyy");
+            if (OrganisationIsRequiredToSubmit() && reportIsNotSubmitted && DeadlineHasPassed())
+            {
+                return ReportStatus.Overdue;
             }
 
-            return null;
+            if (!reportIsNotSubmitted
+                && (returnForYear.IsRequired() && returnForYear.IsSubmittedOnTime()

```

Comment:
We call this quite a bunch of times - what about: `var returnIsRequired = returnForYear.IsRequired()`